test/user_permissions.go

```diff
@@ -0,0 +1,71 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type AllowedOperations struct {
+	Get    bool
+	List   bool
+	Create bool
+	Delete bool
+}
+
+var (
+	AllowAll = AllowedOperations{
+		Get:    true,
+		List:   true,
+		Create: true,
+		Delete: true,
+	}
+	AllowViewOnly = AllowedOperations{
+		Get:  true,
+		List: true,
+	}
+)
+
+type UserPermissionTest struct {
+	Name              string
+	UserContext       *Context
+	AllowedOperations map[schema.GroupVersionResource]AllowedOperations
+}
+
+func RunUserPermissionTests(t *testing.T, objects map[schema.GroupVersionResource]*unstructured.Unstructured, tests ...UserPermissionTest) {
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			for gvr, allowed := range test.AllowedOperations {
+				client := test.UserContext.Clients.Dynamic.Resource(gvr).Namespace("serverless-tests")
```

Comment:
I guess parameterizing is not worth it - nor needed, right?

Reply:
it's the same everywhere currently, yep. I'd parametrize if needed.